api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -75,6 +77,26 @@ public URI getBaseUri() {
 
   @Override
   public void close() {
+    if (client != null) {
+      client
+          .executor()
+          .ifPresent(
+              executor -> {
+                if (executor instanceof ExecutorService) {
```

Comment:
This looks like we're digging into implementation details... How is `HttpClient` supposed to be closed normally? I'd guess its threads should eventually finish when not in use. WDYT?

Reply:
@dimas-b there seems to be couple options: `java.net.http.HttpClient.close()`, `java.net.http.HttpClient.shutdown()`, `java.net.http.HttpClient.shutdownNow()` (apparently copied from ExecutorService)

Reply:
Perhaps we could check if it implements `AutoClosable` (java 21), then cast and call `.close()`... WDYT?

Reply:
Yes, except that you probably want shutdownNow-like behavior over just shutdown. 
Java 21's `java.net.http.HttpClient.close()` will issue shutdown, then wait and then shutdownNow only if interrupted (same as ExecutorService).

Reply:
I have asked for the thread dump which caused Trino folks to report the issue. I will also take a thread dump and analyze tomorrow.

Reply:
If we waited a bit, would the executor release its thread automatically? Could this be a matter of configuring the thread leak detection timeout?